pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -84,12 +84,13 @@ var (
 )
 
 func getKey(ds *extensions.DaemonSet, t *testing.T) string {
-	if key, err := controller.KeyFunc(ds); err != nil {
+	var key string
+	var err error
+	if key, err = controller.KeyFunc(ds); err != nil {
```

Comment:
clearer and less lines of code if you pull the assignement out of the if
```
func getKey(ds *extensions.DaemonSet, t *testing.T) string {
  key, err := ...
  if err != nil {
```

Reply:
Good point.
I've also noticed [MetaNamespaceKeyFunc](https://github.com/kubernetes/kubernetes/blob/23f0fe8632bea74e449617fc95b984155bfc768e/staging/src/k8s.io/client-go/tools/cache/store.go#L82) returns an empty string if there's an error.
There's therefore no need for a double return. We can just always return key.